tests/vnc_test.go

```diff
@@ -166,6 +118,72 @@ var _ = Describe("[rfe_id:127][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			_, err = wrappedRoundTripper.RoundTrip(req)
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		DescribeTable("Dual connection. Validate default behavior and with 'preserve-session' option", func(preserve1, preserve2 bool) {
+			newCtxWithConnect := func() (context.Context, context.CancelFunc, chan struct{}, chan struct{}) {
+				connect := make(chan struct{})
+				disconnect := make(chan struct{})
+				ctx := context.WithValue(context.Background(), connectedKeyVNC, connect)
+				ctx = context.WithValue(ctx, keepConnectedKeyVNC, disconnect)
+				ctx, cancel := context.WithTimeout(ctx, vncWaitResponseTimeout)
+				return ctx, cancel, connect, disconnect
+			}
+
+			ctx1, cancel1, connected1, disconnect1 := newCtxWithConnect()
+			DeferCleanup(cancel1)
+			go func() {
+				defer GinkgoRecover()
+				vncConnect(ctx1, vmi, preserve1, "")
+			}()
+
+			// wait connect
+			select {
+			case <-connected1:
+			case <-ctx1.Done():
+				// Timeout
+				Expect(ctx1.Err()).ToNot(HaveOccurred())
+			}
+
+			ctx2, cancel2, connected2, disconnect2 := newCtxWithConnect()
+			DeferCleanup(cancel2)
+			go func() {
+				defer GinkgoRecover()
+				errMsg := ""
+				if preserve2 {
+					errMsg = "websocket: bad handshake"
+				}
+				vncConnect(ctx2, vmi, preserve2, errMsg)
+			}()
+
+			// Wait connect or timeout
+			select {
+			case <-connected2:
+			case <-ctx2.Done():
+				Expect(preserve2).To(BeTrue())
+				Expect(ctx2.Err()).ToNot(HaveOccurred())
+			case <-time.After(time.Second * 2):
+				Expect(preserve2).To(BeTrue())
+				disconnect2 <- struct{}{}
+				close(disconnect2)
+			}
```

Comment:
Can this be expressed with `Eventually` and the `Receive` matcher instead?

Reply:
One liner each. Super nice.
```go
Eventually(connected1).Should(Receive(pointer.P(true), BeTrue()))
```